redbot/cogs/streams/streamtypes.py

```diff
@@ -68,14 +69,19 @@ async def is_online(self):
     def make_embed(self):
         raise NotImplementedError()
 
+    def iter_messages(self):
+        for msg_data in self.messages:
+            data = msg_data.copy()
+            channel = self.bot.get_channel(msg_data["channel"])
```

Comment:
This should be replaced with `self.bot.get_guild(data.pop("guild_id"))` and `guild.get_channel(msg_data["channel"]) with the addition of the previous comment.